### PR TITLE
Remove rethrowable runtime exception during connection

### PR DESCRIPTION
**Pull request by ajax-surovskyi-y**
https://github.com/nats-io/nats.java/issues/1136

### CODE REVIEW

File: src/main/java/io/nats/client/impl/NatsConnection.java [hunk 1]
```diff
@@ -553,9 +553,6 @@ public void run() {
                 statusLock.unlock();
             }
             timeTraceLogger.trace("status updated");
-        } catch (RuntimeException exp) { // runtime exceptions, like illegalArgs
-            processException(exp);
-            throw exp;
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsConnection.java:558]: This is simply adding the ability to report this message via process exception, which is why it is caught and rethrown. We can revisit this in a different PR if you still feel strongly about this.

**Review by scottf — CHANGES_REQUESTED** [review]
Replace the removal of the runtime exception and I will approve this PR. We can revisit that issue independantly.

**Comment by ajax-surovskyi-y** [comment]
I noticed that in another PR, you have already removed the handling of RuntimeException. Thus, the only remaining task is to remove the unnecessary handling of CancellationException. I am looking forward to contributing to your project.

**Comment by scottf** [comment]
@ajax-surovskyi-y Something is wrong with this PR, it looks like it is repeating stuff just merged from another PR. Please get a fresh copy of main branch and PR into a different branch. I'm closing this PR for that reason only.

**Review by scottf — APPROVED** [review]
LGTM

**Pull request merged**